src/rpc/methods/eth.rs

```diff
@@ -2381,4 +2595,302 @@ mod test {
         let block = Block::new(true, 1);
         assert_eq!(block.transactions_root, EthHash::default());
     }
+
+    #[test]
+    fn test_eth_tx_hash_from_signed_message() {
+        let (_, signed) = construct_eth_messages(0);
+        let tx_hash =
+            eth_tx_hash_from_signed_message(&signed, crate::networks::calibnet::ETH_CHAIN_ID)
+                .unwrap();
+        assert_eq!(
+            &format!("{}", tx_hash),
+            "0xfc81dd8d9ffb045e7e2d494f925824098183263c7f402d69e18cc25e3422791b"
+        );
+
+        let (_, signed) = construct_messages();
+        let tx_hash =
+            eth_tx_hash_from_signed_message(&signed, crate::networks::calibnet::ETH_CHAIN_ID)
+                .unwrap();
+        assert_eq!(tx_hash.to_cid(), signed.cid());
+
+        let (_, signed) = construct_bls_messages();
+        let tx_hash =
+            eth_tx_hash_from_signed_message(&signed, crate::networks::calibnet::ETH_CHAIN_ID)
+                .unwrap();
+        assert_eq!(tx_hash.to_cid(), signed.message().cid());
+    }
+
+    #[test]
+    fn test_eth_tx_hash_from_message_cid() {
+        let blockstore = Arc::new(MemoryDB::default());
+
+        let (msg0, secp0) = construct_eth_messages(0);
+        let (_msg1, secp1) = construct_eth_messages(1);
+        let (msg2, bls0) = construct_bls_messages();
+
+        crate::chain::persist_objects(&blockstore, [msg0.clone(), msg2.clone()].iter()).unwrap();
+        crate::chain::persist_objects(&blockstore, [secp0.clone(), bls0.clone()].iter()).unwrap();
+
+        let tx_hash = eth_tx_hash_from_message_cid(&blockstore, &secp0.cid(), 0).unwrap();
+        assert!(tx_hash.is_some());
+
+        let tx_hash = eth_tx_hash_from_message_cid(&blockstore, &msg2.cid(), 0).unwrap();
+        assert!(tx_hash.is_some());
+
+        let tx_hash = eth_tx_hash_from_message_cid(&blockstore, &secp1.cid(), 0).unwrap();
+        assert!(tx_hash.is_none());
+    }
+
+    #[test]
+    fn test_eth_log_from_event() {
+        let entries = vec![
+            EventEntry {
+                flags: (Flags::FLAG_INDEXED_ALL).bits(),
+                key: "t1".into(),
+                codec: IPLD_RAW,
+                value: vec![
+                    226, 71, 32, 244, 92, 183, 79, 45, 85, 241, 222, 235, 182, 9, 143, 80, 241, 11,
+                    81, 29, 171, 138, 125, 71, 196, 129, 154, 8, 220, 208, 184, 149,
+                ]
```

Comment:
It'd be great to document where those test vectors come from.

Reply:
These payloads correspond to existing topics on Calibnet, but they could just as easily be vectors filled with random bytes, as long as they are 32 bytes.

Reply:
The comment that you added is also okay.